core/src/main/c/windows/files.c

```diff
@@ -238,6 +238,28 @@ JNIEXPORT jlong JNICALL Java_io_questdb_std_Files_length0(JNIEnv *e, jclass cl,
     return len;
 }
 
+JNIEXPORT jint JNICALL Java_io_questdb_std_Files_hardLink(JNIEnv *e, jclass cl, jlong lpszSrc, jlong lpszHardLink) {
+
+    size_t lenSrc = MultiByteToWideChar(CP_UTF8, 0, (LPCCH) lpszSrc, -1, NULL, 0);
+    if (lenSrc > 0) {
+        wchar_t bufSrc[lenSrc];
+        MultiByteToWideChar(CP_UTF8, 0, (LPCCH) lpszSrc, -1, bufSrc, (int) lenSrc);
+
+        size_t lenHardLink = MultiByteToWideChar(CP_UTF8, 0, (LPCCH) lpszHardLink, -1, NULL, 0);
+        if (lenHardLink > 0) {
+            wchar_t bufHardLink[lenHardLink];
+            MultiByteToWideChar(CP_UTF8, 0, (LPCCH) lpszHardLink, -1, bufHardLink, (int) lenHardLink);
+
+            if (CreateHardLinkA(lpszHardLink, lpszSrc, NULL)) {
+                return 0;
+            }
+            SaveLastError();
+            return -1;
+        }
+    }
+    return -2;
```

Comment:
this is also inconsistent with what linux does, which never returns -2. I would suggest to examine name in Java to make sure it is non-empty and assume in C that names are non-zero. When they are zero in C - return -1, which would have no errno set.